### ISSUE TITLE
Player 2/3/4 etc. pocket address offsets

### ISSUE THREAD

**Issue by DanielVip3** [gqxastg/ACNH-Cheat-Tables#1]
Hi!

I have not a good understanding of memory editing, especially with Cheat Engine.
I have 2 players (villagers) in my island, and as you may know, Player 1 and Player 4's address are switched when the island has more than one player. I am trying to edit my 1st villager's inventory, but without success because it actually shows me my 2nd villager's inventory. I need to edit it by using the 4th villager's address.

My question is straightforward: is there a way I can edit the code to let me have the pocket address offset for the 4th player, instead of the 1st? As if I was the 4th player.

I understand little about the sums and subtractions you do with addresses and why, so I don't know what I can do to edit them.
I tried actually the pocket offsets addresses I found [there](https://gbatemp.net/threads/animal-crossing-new-horizons-cheat-database.563423/), but those have nothing in common with the ones I find in your tables.

If you can actually help me, I could also try doing a PR to implement it as a setting, that would be useful to other people too I guess.

**Comment by gqxastg** [comment]
Well, The first thing to note is I actually have never tested in more than one player, because I only play alone.
I'm just a beginner in using Cheat Engine too, I find these address offsets when I test in single player and they just seem to work when I change their values, that's all.

**For your question**, yep? but I can only say it's hard to explain how to search the address.
I think the relative structure of data in memory of different players is the same, all you need to do is use byte arrays or groups to search for some adjacent item data. (though I'm not sure how to check whether the results are correct, maybe you have to switch player after changing value)

About the Cheat Database, They are all based on a (or two?) base address offset, but I didn't specifically update it. (well, I'm still not sure about this)

So what I want to say is that I'd like to solve this problem myself, but I'm afraid it's tough because of my personal needs and computer hardware.

**In short**, You can try to solve it yourself with the above ideas and I will consider it too, but the priority is not high.
_(I actually don't want to create three more characters... even seven more?)_

_oh ... If different numbers of players have different address offsets, I can't help it I guess. idk._

**Comment by DanielVip3** [comment]
I may be too unexperienced, but I don't know what to search when searching for byte arrays or groups. I sometimes am able to find the address for the item type for the first slot of the pocket or things like that, but I don't know how to find about item count and flags.

I thought just adding an offset in one of the base variables that get summed to obtain the actual address (e.g. the base address of ITEM 20 is pocket2+0+8*13, I thought of changing pocket2 or multiplying) could do the job, but it seems trickier.

I have no clue at all on how to do it. Don't worry if you don't have time, I'll try myself experimenting and I'll come up with something.

**Comment by gqxastg** [comment]
**2p:"ITEM ID" 1:"FLAG1" 1:"FLAG2" 2:"COUNT" 2:"USES"**
Scan Type set as Groups, input this sequence repeatedly to search INVENTORY 01-20 (pocket2) or INVENTORY 21-40 (pocket1).
such as:　_2p:2377 1:0 1:0 2:0 2:8 2p:2099 1:0 1:0 2:0 2:41_

![Example](https://source.acexy.cn/view/X13nlUT)

Please note: COUNT is start with 0 (this means the value will be 0 if you have one)
and you can open the CT as document, search the item name to get id.

Once you add a search result, you will get the IDs of all search items, so you can simply change them to check whether they are correct. (if you have opened the inventory screen, you need to reopen it)

To get FLAG or other data of inventory, you can use [NHSE](https://github.com/kwsch/NHSE) ([current latest version](https://dev.azure.com/project-pokemon/NHSE/_build/results?buildId=4456&view=artifacts&pathAsName=false&type=publishedArtifacts)) to check save file (main.dat). 
In NHSE, FLAG1 is Flag0, FLAG2 is Wrapped (unwrapped is 0)

By the way, I can find a total of three groups of results in the case of single player.

I think that's all I can provide.

**Comment by DanielVip3** [comment]
Thanks! You've been really helpful, that's all I needed to know to understand how to scan for those addresses. I think I'm gonna do this job with all the addresses for P1 when P2 is present, and then test it again with another third player to see if addresses change.

For now, I'll scan and come up with a list of pocket addresses for Player 1 when more than 1 player for AC 2.0+.

Many thanks again! I'll keep you updated.
